[Jun Wei] iP

junwei26 opened the pull request:
# Duke
>"Simply amazing" - me ([source](http://google.com))

Duke frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~Fast~~ _SUPER FAST_ to use

All you need to do is,
1. download it from [here] (http://google.com).
2. let it manage your tasks for you 😉

And it is **free**!

Features

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

junwei26 committed:
Level-1 Done

junwei26 committed:
Level-2 done

junwei26 committed:
Level-3 + extension done

junwei26 committed:
Level-4 done

junwei26 committed:
Fixed task descriptions not showing properly

junwei26 committed:
Added Text ui testing

junwei26 committed:
Level-5 complete

junwei26 committed:
Level-6 complete

junwei26 committed:
Add saving and loading of task data

junwei26 committed:
Add handling of dates for deadlines

junwei26 committed:
Simplify parsing of string to date object for deadline

junwei26 committed:
Merge branch 'branch-Level-7'

junwei26 committed:
Merge branch 'branch-Level-8'

junwei26 committed:
Extract out storage class to handle saving and loading data

junwei26 committed:
Extracted out parser to handle commands

junwei26 committed:
Extract Ui and TaskList classes

junwei26 committed:
Extract Commands into seperate classes

junwei26 committed:
Add duke, duke.task and duke.command packages

junwei26 committed:
Add unit tests for todo, event and deadline

junwei26 committed:
Create jar file and published

junwei26 committed:
Add Javadocs for some public methods

junwei26 committed:
Edit code to adhere to coding standard

junwei26 committed:
Add find command to find tasks

junwei26 committed:
Merge branch 'branch-A-JavaDoc'

junwei26 committed:
Merge branch 'branch-A-CodingStandard'

junwei26 committed:
Fix merge conflicts with branch-Level-9

junwei26 committed:
Fix code style after resolving merge conflicts

junwei26 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

junwei26 committed:
Fix code style errors based on checkstyle

knotstoks commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/duke.Duke.java` file, right-click it, and choose `Run duke.Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
Good attention to detail to edit the README!

knotstoks commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,53 @@
+     * Constructs a duke chat bot with the given file path to store and
+     * load data from.
+     *
+     * @param filePath relative file path to data storage
```
Javadoc statements should end in a fullstop, but honestly this is just a small thing

I just need something to write on

knotstoks commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,36 @@
+    private String dukeError;
+
+    /**
+     * Constructor for a DukeException
```
There should be a space between headers and parameter sections in the Javadoc comment

knotstoks commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,122 @@
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+
```
Not sure if the whitespace is here because you want to add anything else but remember to delete it after!

knotstoks commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,122 @@
+        String[] inputArr = input.split(",");
+        Task currentTask;
+        switch (inputArr[0]) {
+        case "T" :
```
For case statements, there shouldn't be a whitespace between "T" and :

Just gonna paste this for reference

```
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```

knotstoks commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,113 @@
+
+    /**
+     * Constructor for AddCommand.
+     * @param userInput String containing the user input
+     * @param taskType String identifying type of task to be added
```
The full stop thing I mentioned earlier, very minor detail so I won't mention it anymore

knotstoks reviewed (commented):
LGTM! Nothing too major or needing of bug fixing, just a few adjustments according to the coding standard set!

💯

junwei26 committed:
Add basic GUI for duke

junwei26 committed:
Merge branch 'branch-level-10'

hpkoh commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,58 @@
+        userInput.clear();
+    }
+
+    private void message() {
```
Perhaps this should be in verb form instead, ```printMessage()``` or ```displayMessage()```?

hpkoh commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,27 @@
+    public String printError(DukeException e) {
+        return printMessage(e.toString());
+    }
+
```
Perhaps these public methods could use some javadocs?

hpkoh commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,42 @@
+
+public class Deadline extends Task {
+
+    protected LocalDate by;
```
Perhaps there's a more descriptive for this field like ```deadlineDate```

hpkoh commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,43 @@
+    public String toString() {
+        return String.format("[%s] %s", this.getStatusIcon(), this.description);
+    }
+
```
Some public methods here that can use javadocs too

hpkoh reviewed (commented):
Overall, the code follows the coding standard very well! I had to really scrutinize to find a few nitpicks. Just a small note, there is some excess white space in some files after the import statements and at the end of file.

junwei26 committed:
Add asserts to Commands

There are several areas in the Commands that require input to be of
a specific format, but no explicit checks are done to confirm this.

This might break the program if nothing is done to check it.

Adding assert statements at critical points of the Command will
ensure that the format is correct.

Assert statements can be disabled and hence will not have any side
effects on the rest of the program.

junwei26 committed:
Refactor code to improve readability

Add Command's methods are too long and it is hard to understand what
is going on.

This might cause Add Command to be hard to maintain as other
developers might not understand it easily.

The original methods are refactored and helper methods are used
to make the code easier to understand.

This is done to allow developers to understand what is going on
easily without needing to understand the underlying implementation.

junwei26 committed:
Merge pull request #2 from junwei26/branch-A-Assertions

Add asserts to Commands

junwei26 committed:
Fix merge conflicts with master

junwei26 committed:
Merge pull request #1 from junwei26/branch-A-CodeQuality

Refactor code to improve readability

junwei26 committed:
Minor fix at assert statements in Add Command

junwei26 committed:
Set up continuous integration using github actions

junwei26 committed:
Fix code style for source and test code

junwei26 committed:
Add ability to update tasks using update command

junwei26 committed:
Merge branch 'branch-C-Update'

junwei26 committed:
Improve GUI

junwei26 committed:
Update User Guide

junwei26 committed:
Fix user guide formatting issue and add list command to user guide
